### PR TITLE
[TESTCASE+Solution] adding test replaceBodyShouldNotBreakAnonymousClasses and fixing it

### PR DESCRIPTION
**Pull request by ftomassetti**
See #1999 . Thank you @signed for providing this

### CODE REVIEW

**Comment by ftomassetti** [comment]
Updated to include a few derived tests and a solution

**Comment by matozoid** [comment]
Well, if it's a testcase+solution, then it's just a normal PR, right? :-)

**Comment by ftomassetti** [comment]
Yes, probably it is but I wanted to underline the fact I took the testcase prepared by someone else

**Comment by matozoid** [comment]
Oh okay. I assume a [label] means "don't pull this PR", so no need for that.

**Comment by matozoid** [comment]
Yay, @ftomassetti, you've got an EOL mistake :-/

File: javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/transformations/ast/body/MethodDeclarationTransformationsTest.java [hunk 1]
```diff
@@ -431,4 +434,40 @@ void removingAnnotationsWithSpaces() {
                         "}\n", result);
     }
 
+    @Test
+    public void parseAndPrintAnonymousClassExpression() {
+        Expression expression = JavaParser.parseExpression("new Object() {" + EOL +
+                "}");
+        String expected = "new Object() {\n" +
```
Comment (ftomassetti) [inline javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/transformations/ast/body/MethodDeclarationTransformationsTest.java]:
```suggestion
         String expected = "new Object() {" + EOL +
```

**Comment by ftomassetti** [comment]
Let's see if this fix it

**Comment by matozoid** [comment]
It did :-)

**Pull request merged**